libs/test-utils/src/expect_print.ts

```diff
@@ -25,6 +26,11 @@ export function resetExpectPrint(): void {
   lastPrintedElement = undefined;
   lastPrintedElementWithCallback = undefined;
   lastPrintOptions = undefined;
+  errorOnNextPrint = undefined;
+}
+
+export function throwOnNextPrint(error: Error = new Error()): void {
```

Comment:
It's to mock the case where printing returns an error (e.g. the printer is not connected, or more errors we may introduce later in this project) and to test our print error handling. Will add a JSDoc comment at a minimum.

Reply:
Ohhh that makes so much more sense. I totally misunderstood what this function did. I thought it was saying "the next time a print occurs, throw an assertion error that makes the test fail" rather than "have the mock print function throw an error the next time it's called"

A comment would certainly help. A couple other ideas to clarify:
- It might be nice to have a more visible idea of the "mock" that underlies this. E.g. maybe calling `printMock.throwOnNextPrint()` would be clearer
- Some other name ideas: `simulateErrorOnNextPrint` or `printMock.throwErrorOnNextCall`

Reply:
Cool, thanks for the ideas. I think I'll do `simulateErrorOnNextPrint` but, later, separately, I might rename everything to `MockPrint.method()` because it would be nice to group / namespace things (not sure what that would be called). Because it is a unified module that even has state.

Reply:
Sounds good! I agree that making the state held by the mock more explicit would be nice down the line